### PR TITLE
US29 - As a user, I would like to show paths optimized for access by students with disabilities + Initial Floor in Indoor Maps selected

### PR DESCRIPTION
**Pull request by mohanadarafe**
# Content of PR
In the following pull request, you will find two main features.

### **Feature 1: US29 - As a user, I would like to show paths optimized for access by students with disabilities.**
Now, the directions for **different floors** is modified. First, you will notice that the bottom menu of the indoor map connects to the preference menu. If **mobility reduced** is selected on _YES_, the user is guided to the elevator. But, if it is selected on _NO_, the user will be guided to the closest stairs (or elevator if it's closer!). 

Below is a preview of the scenario:
ezgif-3-a59d1cdf2099

### **Feature 2: Indoor Map selects the floor they journey starts from**
I know, I took the GIF above prior to implementing this. But, rest assured, for any journey you take, the indoor map will detect where it should start & set the initial floor by itself!

### Todo
Something we can do is add the "next step" arrow as in outdoor directions but inside. So, if you go from H811 to H520, the next arrow will go to floor 5.

### Before reviewing the PR
You will see a lot of `mobility={mobility}` at random places. We are passing it down from a high level to lower level following this structure.

Untitled Diagram (2)

### CODE REVIEW

File: assets/floormaps/hall/HallFloorX.js [hunk 1]
```diff
@@ -8,17 +8,15 @@ export function HallFloorX(props) {
   const [floorNumber, setFloorNumber] = React.useState("");
   const to = props.to;
   const from = props.from;
+  const mobility = props.mobility;
 
   const floorSelected = async () => {
     let name = await AsyncStorage.getItem("floorSelected");
-    setFloorNumber(name);
+    name == 1 ? setFloorNumber(1) : setFloorNumber(name);
```
Comment (AlainJobs) [inline assets/floormaps/hall/HallFloorX.js:15]: is name a number or a value?

File: assets/floormaps/hall/HallFloorX.js [hunk 2]
```diff
@@ -8,17 +8,15 @@ export function HallFloorX(props) {
...
   };
 
   useEffect(() => {
-    const intervalId = setInterval(() => {
```
Comment (AlainJobs) [inline assets/floormaps/hall/HallFloorX.js:18]: I appreciate that.

File: components/BottomMenu.js [hunk 3]
```diff
@@ -51,19 +58,34 @@ function BottomMenu (props) {
...
         props.navigation.navigate("Directions", { destinationResponse: props.directionResponse });
     };
 
-    const goToPreferenceMenu = () => {
-        props.navigation.navigate("PreferenceMenu", {
-            personaType: personaType,
-            mobilityType: mobilityReduced,
-            transportType: methodTravel
-        });
+    const goToPreferenceMenu = (isIndoor) => {
```
Comment (AlainJobs) [inline components/BottomMenu.js:72]: well done!

File: components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js [hunk 4]
```diff
@@ -144,6 +144,33 @@ export const ConvertPointOfInterest = (name) => {
     }
 }
 
+/**
```
Comment (AlainJobs) [inline components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js:147]: ⛷️ 🥇

File: assets/floormaps/hall/HallFloorX.js (hunk 1 above)
Comment (mohanadarafe) [reply assets/floormaps/hall/HallFloorX.js:15]: ill change the name of the variable to `floor`

File: components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js [hunk 5]
```diff
@@ -144,6 +144,33 @@ export const ConvertPointOfInterest = (name) => {
...
+ * The following function returns the shortest path to
```
Comment (bkenza) [inline components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js:148]: Well documented!

File: components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js [hunk 6]
```diff
@@ -144,6 +144,33 @@ export const ConvertPointOfInterest = (name) => {
...
+    var costToSW = dijkstra(graph, from, "stairs_SW").distance;
+    paths.push({id: "elevator", cost: costToElevator}, {id: "stairs_NE", cost: costToNE}, {id: "stairs_NW", cost: costToNW}, {id: "stairs_SE", cost: costToSE}, {id: "stairs_SW", cost: costToSW});
+
+    paths.forEach((element) => {
+        if (element.cost < smallestCost) {
+            smallestCost = element.cost;
+            closestMethod = element.id
+        }
+    });
+    return closestMethod;
```
Comment (csbduzi) [inline components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js:171]: Nice algorithm here!

File: components/BottomMenu.js [hunk 7]
```diff
@@ -3,6 +3,8 @@ import React, { useEffect } from "react";
 import { View, AsyncStorage, Text, StyleSheet, Switch } from "react-native";
 import { Icon } from "native-base";
 import { Button } from "react-native-paper";
+import { FloorMenu } from "./FloorMenu";
+import { getFloorNumber } from "./IndoorDirections/Dijkstra/DijkstraAlgorithm";
```
Comment (Firassawan) [inline components/BottomMenu.js:7]: Screen Shot 2020-04-10 at 3 44 23 PM

File: components/BottomMenu.js [hunk 8]
```diff
@@ -23,7 +25,12 @@ function BottomMenu (props) {
     const [methodTravel, setMethodTravel] = React.useState("");
     const [personaType, setPersonaType] = React.useState("");
     const [mobilityReduced, setMobilityReduced] = React.useState("");
+    const viewIndoor = props.indoor;
+    const inDirections = props.inDirections;
+    const building = props.building;
     const previewDirections = props.previewMode;
+    const from = props.from;
```
Comment (Firassawan) [inline components/BottomMenu.js:32]: Screen Shot 2020-04-10 at 3 44 52 PM

File: components/BottomMenu.js [hunk 9]
```diff
@@ -23,7 +25,12 @@ function BottomMenu (props) {
...
+    const to = props.to;
```
Comment (Firassawan) [inline components/BottomMenu.js:33]: Screen Shot 2020-04-10 at 3 45 06 PM

File: components/BottomMenu.js [hunk 10]
```diff
@@ -111,10 +133,26 @@ function BottomMenu (props) {
         }
     };
 
+    if (viewIndoor) {
+        return(
+            <View style={styles.insideBuildingContainer} data-test="BottomMenu" testID="bottomMenuInitalView">
+                <Icon name="ios-arrow-up" style={styles.arrowUp} onPress={() => {(inDirections ? goToPreferenceMenu(true) : goToMoreDetails)}} />
```
Comment (Firassawan) [inline components/BottomMenu.js:139]: Screen Shot 2020-04-10 at 3 45 33 PM

File: components/FloorMenu.js [hunk 11]
```diff
@@ -39,21 +41,53 @@ const vlFloors = [
...
+    var floor = getFloorNumber(from)
+    AsyncStorage.setItem("floorSelected", floor.toString());
+    return floor;
+  }
+  if(path == "DIFFERENT_BUILDING") {
+    if (from.includes(" ")) {
+      AsyncStorage.setItem("floorSelected", "1");
+      return 1;
+    } else {
+      var floor = getFloorNumber(from)
```
Comment (Firassawan) [inline components/FloorMenu.js:68]: Screen Shot 2020-04-10 at 3 46 13 PM

File: components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js [hunk 12]
```diff
@@ -144,6 +144,33 @@ export const ConvertPointOfInterest = (name) => {
...
+}
+
 /**
  * The following function returns the shortest path to
  * the interest point requested by the user.
```
Comment (Firassawan) [inline components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js:176]: Screen Shot 2020-04-10 at 3 48 00 PM

File: screens/PreferenceMenu.js [hunk 13]
```diff
@@ -56,9 +58,17 @@ function PreferenceMenu (props) {
...
+            personaType: onPressFirstCategory.selectedButton,
+            mobilityType: onPressSecondCategory.selectedButton,
+            transportType: onPressThirdCategory.selectedButton,
+        });
+    };
+
     return (
         <View style={styles.container}>
-            <Icon name="ios-arrow-down" style={styles.arrowDown} onPress={goToPreviewDirections} />
+            <Icon name="ios-arrow-down" style={styles.arrowDown} onPress={() => {(isIndoor ? goToIndoorMap : goToPreviewDirections)}} />
```
Comment (Firassawan) [inline screens/PreferenceMenu.js:71]: Screen Shot 2020-04-10 at 3 52 03 PM

File: components/BottomMenu.js (hunk 10 above)
Comment (mohanadarafe) [reply components/BottomMenu.js:139]: ignore please

File: components/FloorMenu.js (hunk 11 above)
Comment (mohanadarafe) [reply components/FloorMenu.js:68]: ignore please

File: screens/PreferenceMenu.js (hunk 13 above)
Comment (mohanadarafe) [reply screens/PreferenceMenu.js:71]: ignore please

**Pull request merged**